shared/IConsole.cs

```diff
@@ -15,5 +15,7 @@ public interface IConsole
         bool IsInputRedirected { get; }
         bool IsOutputRedirected { get; }
         bool IsErrorRedirected { get; }
+        ConsoleColor ForegroundColor { get; set; }
```

Comment:
I don't think this guidance applies to command line tools. Users never reference any of the assemblies produced in this repo -- they are all `DotNetCliToolReference` packages -- so breaking changes shouldn't matter.